.github/workflows/plugin.yml

```
@@ -0,0 +1,35 @@
+name: Plugin
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+
+jobs:
+  plugin:
+    name: Plugin test with Composer ${{ matrix.composer }}
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        include:
+          - composer: v1
+          - composer: v2
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Setup PHP
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: 7.1
+          tools: composer:${{ matrix.composer }}
+
+      - name: Check Plugin
+        run: |
+          mkdir /tmp/plugin
+          jq '.repositories[0].url="'$(pwd)'"' tests/plugin/composer.json > /tmp/plugin/composer.json
```

is this adding a repositories entry to composer.json? and if yes, why does that change make the plugin install the http factory?

---

`tests/plugin/composer.json` looks like that:

```json

    "repositories": [
        {
            "type": "path",
            "url": "../..",
            "options": {
                "versions": {
                    "php-http/discovery": "99.99.x-dev"
                }
            }
        }
    ],
    "require": {
        "guzzlehttp/psr7": "^1",
        "php-http/discovery": "99.99.x-dev",
        "psr/http-factory-implementation": "*"
    },
    "config": {
        "allow-plugins": {
            "php-http/discovery": true
        }
    }
}
```

The fact that guzzle's psr7 *v1* is already required makes `http-interop/http-factory-guzzle` the best option to provide an implementation of `psr/http-factory-implementation`.

The change made with `jq` is to accommodate composer v1, who doesn't support having the repo url be a parent directory.

---

ah, sorry now i see. its not the composer.json of the library but a test fixture.

```suggestion
          # replace the relative path for the repository url with an absolute path for composer 1 compatibility.
          jq '.repositories[0].url="'$(pwd)'"' tests/plugin/composer.json > /tmp/plugin/composer.json
```